### PR TITLE
DOC-7888 CollectionManager sample and docs

### PR DESCRIPTION
**Pull request by osfameron**
Run with:

  mvn -U exec:java -Dexec.mainClass="CollectionManagerExample" -Dexec.cleanupDaemonThreads=false

NB: the referred exception bug in JCBC-1792 is now marked resolved, I haven't retested with HEAD, but the example follows the expected behaviour.

### CODE REVIEW

File: modules/howtos/examples/CollectionManagerExample.java [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        // tag::create-scope[]
+        try {
+          collectionMgr.createScope("example-scope");
+        }
+        catch (ScopeExistsException e) {
+          System.out.println("Scope already exists");
+        }
+        catch (AuthenticationFailureException e) {
+          System.out.println("User doesn't have required privileges to administer bucket");
```
Comment (dnault) [inline modules/howtos/examples/CollectionManagerExample.java]: @osfameron If the user doesn't have the required permissions, the SDK throws CouchbaseException. We don't have an exception to signal an authorization failure. I'll start a discussion with the team about whether we should add one. In the mean time, should we just omit this catch block?

File: modules/howtos/pages/provisioning-cluster-resources.adoc [hunk 2]
```diff
@@ -98,8 +98,51 @@ The `Flush` operation may fail if the bucket does not have flush enabled, in tha
...
+
+[source,java]
+----
+include::example$CollectionManagerExample.java[tag=drop-collection,indent=0]
+
+include::example$CollectionManagerExample.java[tag=drop-scope,indent=0]
+----
 
+Note that you require at least the https://docs.couchbase.com/server/7.0/learn/security/roles.html#bucket-admin[Bucket Admin] privilege to create or drop a Scope.
+The minimum privilege for managing Collections is https://docs.couchbase.com/server/7.0/learn/security/roles.html#manage-collections-in-scope[Manage Collections in Scope].
```
Comment (dnault) [inline modules/howtos/pages/provisioning-cluster-resources.adoc]: User also needs "Data Reader" (or something similar) or else they can't open the bucket to get the collections manager.

**Review by dnault — CHANGES_REQUESTED** [review]
nice stuff!

File: modules/howtos/examples/CollectionManagerExample.java (hunk 1 above)
Comment (osfameron) [reply modules/howtos/examples/CollectionManagerExample.java]: Thanks, I'll remove for now as you suggest!
(I'd seen the error while getting my login connection wrong, and made an assumption...)

File: modules/howtos/pages/provisioning-cluster-resources.adoc (hunk 2 above)
Comment (osfameron) [reply modules/howtos/pages/provisioning-cluster-resources.adoc]: Thanks! In fact I'd already added data reader in the *code*, but wasn't sure that was the best practice, will document.

**Review by RichardSmedley — APPROVED** [review]
LGTM

**Comment by osfameron** [comment]
> Run with:
> 
> mvn -U exec:java -Dexec.mainClass="CollectionManagerExample" -Dexec.cleanupDaemonThreads=false

Better command, which actually compiles it is:

    mvn compile exec:java -Dexec.mainClass=CollectionManagerExample -Dexec.cleanupDaemonThreads=false

**Pull request merged**